tests/src/JIT/Regression/JitBlue/GitHub_24846/GitHub_24846.cs

```diff
@@ -5,64 +5,54 @@
 using System;
 using System.Runtime.CompilerServices;
 
-public class GitHub_24846
+unsafe public class GitHub_24846
 {
-    public static void Test(byte[] destination, byte[] source)
+    public static void TestCopy(byte* destination, byte* source)
     {
-        Unsafe.CopyBlockUnaligned(ref destination[0], ref source[0], 0);
+        Unsafe.CopyBlockUnaligned(ref *destination, ref *source, 0);
```

Comment:
This is probably not what you intended to do, you need to drop the `ref`.

Reply:
Ah, thanks. I'm totally not used to programming with pointers (actually, I'm pretty much not all that used to actually programming in C#, since I spend most of my time writing C++!)

Reply:
Edit: I spend most of my time writing C code that's compiled with a C++ compiler ;-)